include/class.search.php

```diff
@@ -844,6 +851,39 @@ function update($vars, &$errors=array()) {
         if (!$errors && $this->_config->update($vars, $errors))
             $this->_settings = $this->_criteria = null;
 
+        // Set the TEAM visibility
+        $form = $this->getVisibilityForm($vars);
+        if (!$form->isValid())
+            return false;
+
+        $this->clearFlag(self::FLAG_VISIBLE_TEAM | self::FLAG_PUBLIC);
+        $this->staff_id = $thisstaff->getId();
+        $answers = $form->getClean();
+        switch ($answers['visible_to']) {
+        case 'team':
+            $this->setFlag(self::FLAG_VISIBLE_TEAM);
+            $this->staff_id = $answers['visible_team'];
```

Comment:
I **now** see that you're overloading `staff_id` (ignore above comments) -  But I think this goes to show you how confusing it is... adding access table or `team_id` field is a better solution in my opinion.